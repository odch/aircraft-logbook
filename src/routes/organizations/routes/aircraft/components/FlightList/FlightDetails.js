import React from 'react'
import { useIntl } from 'react-intl'
import TextField from '@material-ui/core/TextField'
import Grid from '@material-ui/core/Grid'
import Divider from '@material-ui/core/Divider'
import { aircraft, flight } from '../../../../../../shapes'
import {
  formatDate,
  formatTime,
  getTimeDiff
} from '../../../../../../util/dates'

const FlightDetails = ({ aircraft, flight }) => {
  const intl = useIntl()
  return (
    <div>
      <Grid container>
        <Grid item xs={12} container>
          <Grid item xs={12} sm={4}>
            {renderField(
              'date',
              formatDate(
                flight.blockOffTime,
                flight.departureAerodrome.timezone
              ),
              intl
            )}
          </Grid>
          <Grid item xs={6} sm={4}>
            {renderField('pilot', getMemberName(flight.pilot), intl)}
          </Grid>
          <Grid item xs={6} sm={4}>
            {renderField('instructor', getMemberName(flight.instructor), intl)}
          </Grid>
        </Grid>
        <Grid item xs={12} container>
          <Grid item xs={12} sm={4}>
            {renderField(
              'departureaerodrome',
              getAerodromeName(flight.departureAerodrome),
              intl
            )}
          </Grid>
          <Grid item xs={6} sm={4}>
            {renderField(
              'blockofftime',
              formatTimeWithUtc(
                flight.blockOffTime,
                flight.departureAerodrome.timezone
              ),
              intl
            )}
          </Grid>
          <Grid item xs={6} sm={4}>
            {renderField(
              'takeofftime',
              formatTimeWithUtc(
                flight.takeOffTime,
                flight.departureAerodrome.timezone
              ),
              intl
            )}
          </Grid>
        </Grid>
        <Grid item xs={12} container>
          <Grid item xs={12} sm={4}>
            {renderField(
              'destinationaerodrome',
              getAerodromeName(flight.destinationAerodrome),
              intl
            )}
          </Grid>
          <Grid item xs={6} sm={4}>
            {renderField(
              'blockontime',
              formatTimeWithUtc(
                flight.blockOnTime,
                flight.destinationAerodrome.timezone
              ),
              intl
            )}
          </Grid>
          <Grid item xs={6} sm={4}>
            {renderField(
              'landingtime',
              formatTimeWithUtc(
                flight.landingTime,
                flight.destinationAerodrome.timezone
              ),
              intl
            )}
          </Grid>
        </Grid>
        <Grid item xs={12} container>
          <Grid item xs={12} sm={4}>
            {renderField('nature', getFlightNature(flight.nature, intl), intl)}
          </Grid>
          <Grid item xs={12} sm={4}>
            {renderField(
              'fueluplift',
              getFuel(
                flight.fuelUplift,
                flight.fuelType,
                aircraft.settings.fuelTypes
              ),
              intl
            )}
          </Grid>
          <Grid item xs={12} sm={4}>
            <Grid item xs={12} sm={4}>
              {renderField('oiluplift', getOil(flight.oilUplift), intl)}
            </Grid>
          </Grid>
        </Grid>
        <Grid item xs={12} container>
          <Grid item xs={12} sm={4}>
            {renderField('landings', flight.landings, intl)}
          </Grid>
          <Grid item xs={12} sm={4}>
            {renderField('personsonboard', flight.personsOnBoard || '-', intl)}
          </Grid>
          <Grid item xs={12} sm={8}>
            {renderField('remarks', flight.remarks || '-', intl, true)}
          </Grid>
        </Grid>
        <Grid item xs={12} container>
          <Grid item xs={6} sm={4}>
            {renderField(
              'flighthours_time',
              getTimeDiff(flight.takeOffTime, flight.landingTime),
              intl
            )}
          </Grid>
          <Grid item xs={6} sm={4}>
            {renderField(
              'blockhours',
              getTimeDiff(flight.blockOffTime, flight.blockOnTime),
              intl
            )}
          </Grid>
        </Grid>
        {flight.counters && (
          <Grid item xs={12} container>
            {flight.counters.flightTimeCounter && (
              <Grid item xs={6} sm={4}>
                {renderField(
                  'flighthours_counter',
                  getDecimalRange(flight.counters.flightTimeCounter),
                  intl
                )}
              </Grid>
            )}
            {flight.counters.engineTimeCounter && (
              <Grid item xs={6} sm={4}>
                {aircraft.settings.engineHoursCounterEnabled &&
                  renderField(
                    'enginehours',
                    getDecimalRange(flight.counters.engineTimeCounter),
                    intl
                  )}
              </Grid>
            )}
          </Grid>
        )}
      </Grid>
      <Divider />
      <Grid item xs={12} container>
        <Grid item xs={6} sm={4}>
          {renderField(
            'total.flighthours',
            format2Decimals(flight.counters.flightHours.end),
            intl
          )}
        </Grid>
        {flight.counters.engineHours && (
          <Grid item xs={6} sm={4}>
            {renderField(
              'total.enginehours',
              format2Decimals(flight.counters.engineHours.end),
              intl
            )}
          </Grid>
        )}
        <Grid item xs={6} sm={4}>
          {renderField('total.landings', flight.counters.landings.end, intl)}
        </Grid>
      </Grid>
    </div>
  )
}

const renderField = (label, value, intl, multiline = false) => (
  <TextField
    label={intl.formatMessage({ id: `flightlist.${label}` })}
    value={value}
    margin="normal"
    InputProps={{
      readOnly: true,
      disableUnderline: true
    }}
    fullWidth
    multiline={multiline}
  />
)

const getMemberName = member =>
  member ? `${member.firstname} ${member.lastname}` : '-'

const getAerodromeName = aerodrome =>
  `${aerodrome.identification} (${aerodrome.name})`

const getFlightNature = (nature, intl) =>
  intl.formatMessage({ id: `flight.nature.${nature}` })

const getFuel = (fuelUplift, fuelType, aircraftFuelTypes) =>
  fuelUplift && fuelUplift > 0
    ? `${fuelUplift}L (${getFuelTypeDescription(fuelType, aircraftFuelTypes)})`
    : '-'

const getFuelTypeDescription = (fuelType, aircraftFuelTypes) => {
  const foundType = aircraftFuelTypes.find(type => type.name === fuelType)
  if (foundType && foundType.description) {
    return foundType.description
  }
  return fuelType
}

const getOil = oilUplift => (oilUplift && oilUplift > 0 ? `${oilUplift}L` : '-')

const getDecimalRange = range => {
  const diff = range.end - range.start

  const startFormatted = format2Decimals(range.start)
  const endFormatted = format2Decimals(range.end)
  const diffFormatted = format2Decimals(diff)

  return `${diffFormatted} (${startFormatted} - ${endFormatted})`
}

const format2Decimals = value => Number(value / 100).toFixed(2)

const formatTimeWithUtc = (timestamp, timezone) =>
  `${formatTime(timestamp, timezone)} (${formatTime(timestamp, 'UTC')} UTC)`

FlightDetails.propTypes = {
  aircraft,
  flight
}

export default FlightDetails
