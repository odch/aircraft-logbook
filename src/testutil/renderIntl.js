import React from 'react'
import renderer from 'react-test-renderer'
import { createMount } from '@material-ui/core/test-utils'
import { IntlProvider } from 'react-intl'
import pretty from 'pretty'
import messages from '../messages'

const LOCALE = 'de'

const renderInIntlProvider = component => (
  <IntlProvider locale={LOCALE} messages={messages[LOCALE]}>
    {component}
  </IntlProvider>
)

const renderIntl = (component, options) =>
  renderer.create(renderInIntlProvider(component), options)

export const renderIntlUpdate = (renderer, component, options) =>
  renderer.update(renderInIntlProvider(component), options)

export const renderIntlMaterial = (component, returnPrettyHtml = false) => {
  const rendered = createMount()(renderInIntlProvider(component))
  return returnPrettyHtml ? pretty(rendered.html()) : rendered
}

export default renderIntl
