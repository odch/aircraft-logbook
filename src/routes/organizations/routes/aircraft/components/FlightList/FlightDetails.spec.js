import React from 'react'
import renderIntl from '../../../../../../testutil/renderIntl'
import FlightDetails from './FlightDetails'

const counter = (start, end) => ({ start, end })

describe('routes', () => {
  describe('organizations', () => {
    describe('routes', () => {
      describe('aircraft', () => {
        describe('components', () => {
          describe('FlightList', () => {
            describe('FlightDetails', () => {
              const aircraft = {
                id: 'o7flC7jw8jmkOfWo8oyA',
                registration: 'HBKFW',
                settings: {
                  fuelTypes: [
                    {
                      name: 'mogas_homebase',
                      description: 'MoGas (Homebase)'
                    }
                  ],
                  engineHoursCounterEnabled: true
                }
              }

              const flight = {
                id: 'sStfyLd2XArT7oUZPFDn',
                departureAerodrome: {
                  name: 'Lommis',
                  identification: 'LSZT',
                  timezone: 'Europe/Zurich'
                },
                destinationAerodrome: {
                  name: 'Lommis',
                  identification: 'LSZT',
                  timezone: 'Europe/Zurich'
                },
                blockOffTime: {
                  toDate: () => Date.parse('2018-11-20 10:00 GMT+0100')
                },
                blockOnTime: {
                  toDate: () => Date.parse('2018-11-20 11:10 GMT+0100')
                },
                takeOffTime: {
                  toDate: () => Date.parse('2018-11-20 10:10 GMT+0100')
                },
                landingTime: {
                  toDate: () => Date.parse('2018-11-20 11:00 GMT+0100')
                },
                pilot: {
                  firstname: 'Max',
                  lastname: 'Muster'
                },
                instructor: {
                  firstname: 'Hans',
                  lastname: 'Keller'
                },
                counters: {
                  flightHours: counter(10156, 10243),
                  engineHours: counter(10489, 10604),
                  landings: counter(234, 235)
                },
                nature: 'vp',
                landings: 1,
                personsOnBoard: 3,
                remarks: 'my\ntest\nremarks',
                fuelUplift: 58.68,
                fuelType: 'mogas_homebase',
                oilUplift: 0.7,
                troublesObservations: 'troubles',
                techlogEntryStatus: 'not_airworthy',
                techlogEntryDescription: 'loose screw left main wheel'
              }

              const renderToJson = component =>
                renderIntl(component, {
                  createNodeMock: element => {
                    if (element.type === 'textarea') {
                      // mock for TextareaAutosize
                      return document.createElement('textarea')
                    }
                    return null
                  }
                }).toJSON()

              it('renders correctly', () => {
                const tree = renderToJson(
                  <FlightDetails aircraft={aircraft} flight={flight} />
                )
                expect(tree).toMatchSnapshot()
              })

              it('does not render engine hours if counter not enabled', () => {
                const testAircraft = {
                  ...aircraft,
                  settings: {
                    ...aircraft.settings,
                    engineHoursCounterEnabled: false
                  }
                }
                const tree = renderToJson(
                  <FlightDetails aircraft={testAircraft} flight={flight} />
                )
                expect(tree).toMatchSnapshot()
              })
            })
          })
        })
      })
    })
  })
})
