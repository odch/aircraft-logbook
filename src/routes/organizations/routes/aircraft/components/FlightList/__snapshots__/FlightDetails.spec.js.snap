// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`routes organizations routes aircraft components FlightList FlightDetails renders correctly 1`] = `
<div
  className="MuiGrid-container-1"
>
  <div
    className="MuiGrid-container-1 MuiGrid-item-2 MuiGrid-grid-xs-12-41"
  >
    <div
      className="MuiGrid-item-2 MuiGrid-grid-xs-12-41 MuiGrid-grid-sm-4-47"
    >
      <div
        className="MuiFormControl-root-98 MuiFormControl-marginNormal-99 MuiFormControl-fullWidth-101"
      >
        <label
          className="MuiFormLabel-root-109 MuiFormLabel-filled-113 MuiInputLabel-root-102 MuiInputLabel-formControl-103 MuiInputLabel-animated-106 MuiInputLabel-shrink-105"
          data-shrink={true}
        >
          Datum
        </label>
        <div
          className="MuiInputBase-root-129 MuiInput-root-116 MuiInputBase-fullWidth-138 MuiInput-fullWidth-123 MuiInputBase-formControl-130 MuiInput-formControl-117"
          onClick={[Function]}
        >
          <input
            aria-invalid={false}
            className="MuiInputBase-input-139 MuiInput-input-124"
            disabled={false}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            readOnly={true}
            required={false}
            type="text"
            value="20.11.2018"
          />
        </div>
      </div>
    </div>
    <div
      className="MuiGrid-item-2 MuiGrid-grid-xs-6-35 MuiGrid-grid-sm-4-47"
    >
      <div
        className="MuiFormControl-root-98 MuiFormControl-marginNormal-99 MuiFormControl-fullWidth-101"
      >
        <label
          className="MuiFormLabel-root-109 MuiFormLabel-filled-113 MuiInputLabel-root-102 MuiInputLabel-formControl-103 MuiInputLabel-animated-106 MuiInputLabel-shrink-105"
          data-shrink={true}
        >
          Pilot
        </label>
        <div
          className="MuiInputBase-root-129 MuiInput-root-116 MuiInputBase-fullWidth-138 MuiInput-fullWidth-123 MuiInputBase-formControl-130 MuiInput-formControl-117"
          onClick={[Function]}
        >
          <input
            aria-invalid={false}
            className="MuiInputBase-input-139 MuiInput-input-124"
            disabled={false}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            readOnly={true}
            required={false}
            type="text"
            value="Max Muster"
          />
        </div>
      </div>
    </div>
    <div
      className="MuiGrid-item-2 MuiGrid-grid-xs-6-35 MuiGrid-grid-sm-4-47"
    >
      <div
        className="MuiFormControl-root-98 MuiFormControl-marginNormal-99 MuiFormControl-fullWidth-101"
      >
        <label
          className="MuiFormLabel-root-109 MuiFormLabel-filled-113 MuiInputLabel-root-102 MuiInputLabel-formControl-103 MuiInputLabel-animated-106 MuiInputLabel-shrink-105"
          data-shrink={true}
        >
          Fluglehrer
        </label>
        <div
          className="MuiInputBase-root-129 MuiInput-root-116 MuiInputBase-fullWidth-138 MuiInput-fullWidth-123 MuiInputBase-formControl-130 MuiInput-formControl-117"
          onClick={[Function]}
        >
          <input
            aria-invalid={false}
            className="MuiInputBase-input-139 MuiInput-input-124"
            disabled={false}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            readOnly={true}
            required={false}
            type="text"
            value="Hans Keller"
          />
        </div>
      </div>
    </div>
  </div>
  <div
    className="MuiGrid-container-1 MuiGrid-item-2 MuiGrid-grid-xs-12-41"
  >
    <div
      className="MuiGrid-item-2 MuiGrid-grid-xs-12-41 MuiGrid-grid-sm-4-47"
    >
      <div
        className="MuiFormControl-root-98 MuiFormControl-marginNormal-99 MuiFormControl-fullWidth-101"
      >
        <label
          className="MuiFormLabel-root-109 MuiFormLabel-filled-113 MuiInputLabel-root-102 MuiInputLabel-formControl-103 MuiInputLabel-animated-106 MuiInputLabel-shrink-105"
          data-shrink={true}
        >
          Startflugplatz
        </label>
        <div
          className="MuiInputBase-root-129 MuiInput-root-116 MuiInputBase-fullWidth-138 MuiInput-fullWidth-123 MuiInputBase-formControl-130 MuiInput-formControl-117"
          onClick={[Function]}
        >
          <input
            aria-invalid={false}
            className="MuiInputBase-input-139 MuiInput-input-124"
            disabled={false}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            readOnly={true}
            required={false}
            type="text"
            value="Lommis (LSZT)"
          />
        </div>
      </div>
    </div>
    <div
      className="MuiGrid-item-2 MuiGrid-grid-xs-6-35 MuiGrid-grid-sm-4-47"
    >
      <div
        className="MuiFormControl-root-98 MuiFormControl-marginNormal-99 MuiFormControl-fullWidth-101"
      >
        <label
          className="MuiFormLabel-root-109 MuiFormLabel-filled-113 MuiInputLabel-root-102 MuiInputLabel-formControl-103 MuiInputLabel-animated-106 MuiInputLabel-shrink-105"
          data-shrink={true}
        >
          Block off
        </label>
        <div
          className="MuiInputBase-root-129 MuiInput-root-116 MuiInputBase-fullWidth-138 MuiInput-fullWidth-123 MuiInputBase-formControl-130 MuiInput-formControl-117"
          onClick={[Function]}
        >
          <input
            aria-invalid={false}
            className="MuiInputBase-input-139 MuiInput-input-124"
            disabled={false}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            readOnly={true}
            required={false}
            type="text"
            value="10:00"
          />
        </div>
      </div>
    </div>
    <div
      className="MuiGrid-item-2 MuiGrid-grid-xs-6-35 MuiGrid-grid-sm-4-47"
    >
      <div
        className="MuiFormControl-root-98 MuiFormControl-marginNormal-99 MuiFormControl-fullWidth-101"
      >
        <label
          className="MuiFormLabel-root-109 MuiFormLabel-filled-113 MuiInputLabel-root-102 MuiInputLabel-formControl-103 MuiInputLabel-animated-106 MuiInputLabel-shrink-105"
          data-shrink={true}
        >
          Take off
        </label>
        <div
          className="MuiInputBase-root-129 MuiInput-root-116 MuiInputBase-fullWidth-138 MuiInput-fullWidth-123 MuiInputBase-formControl-130 MuiInput-formControl-117"
          onClick={[Function]}
        >
          <input
            aria-invalid={false}
            className="MuiInputBase-input-139 MuiInput-input-124"
            disabled={false}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            readOnly={true}
            required={false}
            type="text"
            value="10:10"
          />
        </div>
      </div>
    </div>
  </div>
  <div
    className="MuiGrid-container-1 MuiGrid-item-2 MuiGrid-grid-xs-12-41"
  >
    <div
      className="MuiGrid-item-2 MuiGrid-grid-xs-12-41 MuiGrid-grid-sm-4-47"
    >
      <div
        className="MuiFormControl-root-98 MuiFormControl-marginNormal-99 MuiFormControl-fullWidth-101"
      >
        <label
          className="MuiFormLabel-root-109 MuiFormLabel-filled-113 MuiInputLabel-root-102 MuiInputLabel-formControl-103 MuiInputLabel-animated-106 MuiInputLabel-shrink-105"
          data-shrink={true}
        >
          Zielflugplatz
        </label>
        <div
          className="MuiInputBase-root-129 MuiInput-root-116 MuiInputBase-fullWidth-138 MuiInput-fullWidth-123 MuiInputBase-formControl-130 MuiInput-formControl-117"
          onClick={[Function]}
        >
          <input
            aria-invalid={false}
            className="MuiInputBase-input-139 MuiInput-input-124"
            disabled={false}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            readOnly={true}
            required={false}
            type="text"
            value="Lommis (LSZT)"
          />
        </div>
      </div>
    </div>
    <div
      className="MuiGrid-item-2 MuiGrid-grid-xs-6-35 MuiGrid-grid-sm-4-47"
    >
      <div
        className="MuiFormControl-root-98 MuiFormControl-marginNormal-99 MuiFormControl-fullWidth-101"
      >
        <label
          className="MuiFormLabel-root-109 MuiFormLabel-filled-113 MuiInputLabel-root-102 MuiInputLabel-formControl-103 MuiInputLabel-animated-106 MuiInputLabel-shrink-105"
          data-shrink={true}
        >
          Landung
        </label>
        <div
          className="MuiInputBase-root-129 MuiInput-root-116 MuiInputBase-fullWidth-138 MuiInput-fullWidth-123 MuiInputBase-formControl-130 MuiInput-formControl-117"
          onClick={[Function]}
        >
          <input
            aria-invalid={false}
            className="MuiInputBase-input-139 MuiInput-input-124"
            disabled={false}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            readOnly={true}
            required={false}
            type="text"
            value="11:00"
          />
        </div>
      </div>
    </div>
    <div
      className="MuiGrid-item-2 MuiGrid-grid-xs-6-35 MuiGrid-grid-sm-4-47"
    >
      <div
        className="MuiFormControl-root-98 MuiFormControl-marginNormal-99 MuiFormControl-fullWidth-101"
      >
        <label
          className="MuiFormLabel-root-109 MuiFormLabel-filled-113 MuiInputLabel-root-102 MuiInputLabel-formControl-103 MuiInputLabel-animated-106 MuiInputLabel-shrink-105"
          data-shrink={true}
        >
          Block on
        </label>
        <div
          className="MuiInputBase-root-129 MuiInput-root-116 MuiInputBase-fullWidth-138 MuiInput-fullWidth-123 MuiInputBase-formControl-130 MuiInput-formControl-117"
          onClick={[Function]}
        >
          <input
            aria-invalid={false}
            className="MuiInputBase-input-139 MuiInput-input-124"
            disabled={false}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            readOnly={true}
            required={false}
            type="text"
            value="11:10"
          />
        </div>
      </div>
    </div>
  </div>
  <div
    className="MuiGrid-container-1 MuiGrid-item-2 MuiGrid-grid-xs-12-41"
  >
    <div
      className="MuiGrid-item-2 MuiGrid-grid-xs-6-35 MuiGrid-grid-sm-4-47"
    >
      <div
        className="MuiFormControl-root-98 MuiFormControl-marginNormal-99 MuiFormControl-fullWidth-101"
      >
        <label
          className="MuiFormLabel-root-109 MuiFormLabel-filled-113 MuiInputLabel-root-102 MuiInputLabel-formControl-103 MuiInputLabel-animated-106 MuiInputLabel-shrink-105"
          data-shrink={true}
        >
          Flugstunden
        </label>
        <div
          className="MuiInputBase-root-129 MuiInput-root-116 MuiInputBase-fullWidth-138 MuiInput-fullWidth-123 MuiInputBase-formControl-130 MuiInput-formControl-117"
          onClick={[Function]}
        >
          <input
            aria-invalid={false}
            className="MuiInputBase-input-139 MuiInput-input-124"
            disabled={false}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            readOnly={true}
            required={false}
            type="text"
            value="101.56 - 102.43"
          />
        </div>
      </div>
    </div>
    <div
      className="MuiGrid-item-2 MuiGrid-grid-xs-6-35 MuiGrid-grid-sm-4-47"
    >
      <div
        className="MuiFormControl-root-98 MuiFormControl-marginNormal-99 MuiFormControl-fullWidth-101"
      >
        <label
          className="MuiFormLabel-root-109 MuiFormLabel-filled-113 MuiInputLabel-root-102 MuiInputLabel-formControl-103 MuiInputLabel-animated-106 MuiInputLabel-shrink-105"
          data-shrink={true}
        >
          Motorstunden
        </label>
        <div
          className="MuiInputBase-root-129 MuiInput-root-116 MuiInputBase-fullWidth-138 MuiInput-fullWidth-123 MuiInputBase-formControl-130 MuiInput-formControl-117"
          onClick={[Function]}
        >
          <input
            aria-invalid={false}
            className="MuiInputBase-input-139 MuiInput-input-124"
            disabled={false}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            readOnly={true}
            required={false}
            type="text"
            value="104.89 - 106.04"
          />
        </div>
      </div>
    </div>
  </div>
  <div
    className="MuiGrid-container-1 MuiGrid-item-2 MuiGrid-grid-xs-12-41"
  >
    <div
      className="MuiGrid-item-2 MuiGrid-grid-xs-12-41 MuiGrid-grid-sm-4-47"
    >
      <div
        className="MuiFormControl-root-98 MuiFormControl-marginNormal-99 MuiFormControl-fullWidth-101"
      >
        <label
          className="MuiFormLabel-root-109 MuiFormLabel-filled-113 MuiInputLabel-root-102 MuiInputLabel-formControl-103 MuiInputLabel-animated-106 MuiInputLabel-shrink-105"
          data-shrink={true}
        >
          Typ
        </label>
        <div
          className="MuiInputBase-root-129 MuiInput-root-116 MuiInputBase-fullWidth-138 MuiInput-fullWidth-123 MuiInputBase-formControl-130 MuiInput-formControl-117"
          onClick={[Function]}
        >
          <input
            aria-invalid={false}
            className="MuiInputBase-input-139 MuiInput-input-124"
            disabled={false}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            readOnly={true}
            required={false}
            type="text"
            value="Charter VFR"
          />
        </div>
      </div>
    </div>
    <div
      className="MuiGrid-item-2 MuiGrid-grid-xs-12-41 MuiGrid-grid-sm-4-47"
    >
      <div
        className="MuiFormControl-root-98 MuiFormControl-marginNormal-99 MuiFormControl-fullWidth-101"
      >
        <label
          className="MuiFormLabel-root-109 MuiFormLabel-filled-113 MuiInputLabel-root-102 MuiInputLabel-formControl-103 MuiInputLabel-animated-106 MuiInputLabel-shrink-105"
          data-shrink={true}
        >
          Treibstoff
        </label>
        <div
          className="MuiInputBase-root-129 MuiInput-root-116 MuiInputBase-fullWidth-138 MuiInput-fullWidth-123 MuiInputBase-formControl-130 MuiInput-formControl-117"
          onClick={[Function]}
        >
          <input
            aria-invalid={false}
            className="MuiInputBase-input-139 MuiInput-input-124"
            disabled={false}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            readOnly={true}
            required={false}
            type="text"
            value="58.68L (MoGas (Homebase))"
          />
        </div>
      </div>
    </div>
    <div
      className="MuiGrid-item-2 MuiGrid-grid-xs-12-41 MuiGrid-grid-sm-4-47"
    >
      <div
        className="MuiGrid-item-2 MuiGrid-grid-xs-12-41 MuiGrid-grid-sm-4-47"
      >
        <div
          className="MuiFormControl-root-98 MuiFormControl-marginNormal-99 MuiFormControl-fullWidth-101"
        >
          <label
            className="MuiFormLabel-root-109 MuiFormLabel-filled-113 MuiInputLabel-root-102 MuiInputLabel-formControl-103 MuiInputLabel-animated-106 MuiInputLabel-shrink-105"
            data-shrink={true}
          >
            Öl
          </label>
          <div
            className="MuiInputBase-root-129 MuiInput-root-116 MuiInputBase-fullWidth-138 MuiInput-fullWidth-123 MuiInputBase-formControl-130 MuiInput-formControl-117"
            onClick={[Function]}
          >
            <input
              aria-invalid={false}
              className="MuiInputBase-input-139 MuiInput-input-124"
              disabled={false}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              readOnly={true}
              required={false}
              type="text"
              value="0.7L"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="MuiGrid-container-1 MuiGrid-item-2 MuiGrid-grid-xs-12-41"
  >
    <div
      className="MuiGrid-item-2 MuiGrid-grid-xs-12-41 MuiGrid-grid-sm-4-47"
    >
      <div
        className="MuiFormControl-root-98 MuiFormControl-marginNormal-99 MuiFormControl-fullWidth-101"
      >
        <label
          className="MuiFormLabel-root-109 MuiFormLabel-filled-113 MuiInputLabel-root-102 MuiInputLabel-formControl-103 MuiInputLabel-animated-106 MuiInputLabel-shrink-105"
          data-shrink={true}
        >
          Landungen
        </label>
        <div
          className="MuiInputBase-root-129 MuiInput-root-116 MuiInputBase-fullWidth-138 MuiInput-fullWidth-123 MuiInputBase-formControl-130 MuiInput-formControl-117"
          onClick={[Function]}
        >
          <input
            aria-invalid={false}
            className="MuiInputBase-input-139 MuiInput-input-124"
            disabled={false}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            readOnly={true}
            required={false}
            type="text"
            value={1}
          />
        </div>
      </div>
    </div>
    <div
      className="MuiGrid-item-2 MuiGrid-grid-xs-12-41 MuiGrid-grid-sm-8-51"
    >
      <div
        className="MuiFormControl-root-98 MuiFormControl-marginNormal-99 MuiFormControl-fullWidth-101"
      >
        <label
          className="MuiFormLabel-root-109 MuiFormLabel-filled-113 MuiInputLabel-root-102 MuiInputLabel-formControl-103 MuiInputLabel-animated-106 MuiInputLabel-shrink-105"
          data-shrink={true}
        >
          Bemerkungen
        </label>
        <div
          className="MuiInputBase-root-129 MuiInput-root-116 MuiInputBase-fullWidth-138 MuiInput-fullWidth-123 MuiInputBase-formControl-130 MuiInput-formControl-117"
          onClick={[Function]}
        >
          <input
            aria-invalid={false}
            className="MuiInputBase-input-139 MuiInput-input-124"
            disabled={false}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            readOnly={true}
            required={false}
            type="text"
            value="my
test
remarks"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
